### PR TITLE
drop outdated guides and merge developer guide

### PR DESCRIPTION
**Pull request by phoracek**
**What this PR does / why we need it**:

Our design documents are outdated. It is hard to keep them up to date
while the project is evolving so fast. For now, let's just drop them
and replace them by a proper design document later.

This patch also merges developer-guide-commands.md into the CONTRIBUTION
document in order to keep all the needed info in a single place and hopefully
make it more visible and easier to keep up to date.

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,54 +1,124 @@
...
+See [local virtualized cluster guide](docs/deployment-local-cluster.md) to learn
+how to deploy a cluster that can be used to try kubernetes-nmstate, run e2e
+tests and perform debugging.
+
+## Building
+
+```shell
+# If pkg/apis/ has been changed, run generator to update client code
+make gen-k8s gen-openapi
+
```
Comment (qinqon) [inline CONTRIBUTING.md]: No longer needed now make handler calls it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,54 +1,124 @@
...
 
-The bot may also make some helpful suggestions for commands to run in your PR to facilitate review.
-These /command options can be entered in comments to trigger auto-labeling and notifications. Refer to its command reference documentation.
+The first difference you'll see is that a bot will begin applying structured
+labels to your PR.
+
+The bot may also make some helpful suggestions for commands to run in your PR to
+facilitate review. These `/command` options can be entered in comments to
+trigger auto-labeling and notifications. Refer to its command reference
+documentation.
```
Comment (qinqon) [inline CONTRIBUTING.md]: Include bot commands list [1] also prow PR process  [2]

[1] https://go.k8s.io/bot-commands?repo=nmstate%2Fkubernetes-nmstate
[2] https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,54 +1,124 @@
...
-    You can "sign" this certificate by including a line in the git commit of "Signed-off-by: Legal Name <email-address>".
-    If you forgot to add the sign-off, you can also amend your commit with the sign-off:
-    `git commit --amend -s`.
+- Missing DCO sign-off:\
+  Developers Certificate of Origin (DCO) Sign-off is a requirement for getting
+  patches into the project (see [Developers Certificate of
+  Origin](https://developercertificate.org/)). You can "sign" this certificate
+  by including a line in the git commit of "Signed-off-by: Legal Name
+  <email-address>". If you forgot to add the sign-off, you can also amend your
+  commit with the sign-off: `git commit --amend -s`.
```
Comment (qinqon) [inline CONTRIBUTING.md]: The git config command to have this automatic would be nice too.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,54 +1,124 @@
...
+need you to:
 
-- Follow the project [coding conventions](https://github.com/golang/go/wiki/CodeReviewComments).
+- Follow the project [coding
+  conventions](https://github.com/golang/go/wiki/CodeReviewComments).
 - Write [good commit messages](https://chris.beams.io/posts/git-commit/).
-- Break large changes into a logical series of smaller patches which individually make easily understandable changes, and in aggregate solve a broader issue.
+- Break large changes into a logical series of smaller patches which
+  individually make easily understandable changes, and in aggregate solve a
+  broader issue.
```
Comment (qinqon) [inline CONTRIBUTING.md]: Also state that we squash PRs, and that multi commit PRs are not that helpful.

File: CONTRIBUTING.md (hunk 2 above)
Comment (phoracek) [reply CONTRIBUTING.md]: Not a part of this PR, I'm just aligning the existing documentation in a narrow block, breaking lines.

File: CONTRIBUTING.md (hunk 3 above)
Comment (phoracek) [reply CONTRIBUTING.md]: Ditto

File: CONTRIBUTING.md (hunk 4 above)
Comment (phoracek) [reply CONTRIBUTING.md]: Ditto

It is helpful for reviews though

File: CONTRIBUTING.md (hunk 1 above)
Comment (phoracek) [reply CONTRIBUTING.md]: It is still handy. e.g. when you adjust API, you need to generate first to start coding and get proper help from your IDE

**Pull request merged**